### ISSUE TITLE
Subtracting out the 'wait time' on the state

### ISSUE THREAD

**Issue by ktwb** [andystewart999/ha_transportnsw#6]
Hi mate,

I've been trying to get this to show in a table the 'time to leaving', but the return of any given sensor is almost always (by way of state) the time until the train(or service as it may be) leaves its origin. 

Is there a way to call this?

**Comment by andystewart999** [comment]
Do you mean that you want to see what the _actual_ departure time is?  If so, that info is in the ```departure_time``` attribute.

Template: ```{{ state_attr('sensor.sensor_name', 'departure_time') }}```
Result: ```2024-10-22T06:20:00```

![image](https://github.com/user-attachments/assets/8e92b362-a76f-4937-b405-961057661aa8)

**Comment by ktwb** [comment]
Not quite - the issue is that the "state" data is the actual departure time. When I feed in the 'wait time', that is when it stops tracking that service and returns the new, latest service.

![image](https://github.com/user-attachments/assets/4e00f4d2-6989-4afe-9d53-787efee87100)


This is fine, however, I was wondering if there is a way to get the 'leaving time' instead of the actual departure time.

I apologise if this is a bit of a weird request - I'm a learner to the world of code/yaml/jinga2 and have probably missed something....

**Comment by ktwb** [comment]
Makes sense. The ultimate use case here is a little sign next to my door which tells me how long until leaving the station, which uses a flex card to order multiple possible train trips. 


The import of that is that

**Comment by andystewart999** [comment]
Good luck - I'm not familar with the card you're using but if it's [this one](https://github.com/custom-cards/flex-table-card), a quick look [here](https://github.com/custom-cards/flex-table-card/blob/master/docs/example-cfg-data.md) suggests that you can't use a template for a column value directly, it would have to be via the template sensor approach discussed.

**Comment by ktwb** [comment]
Alright - I've tried to make the changes. I'm not even sure how Github works to be perfectly honest with you so don't put any weight on how it is that I've clicked 'submit' - I had intended for it to be drafts...

I'm not entirely sure how to test the changes I've made either on my home assistant...

`{% set wiggle_room_in_minutes = 15 %}
Minutes before I should leave for the station: {{ (as_timestamp(state_attr('sensor.arncliffe_to_city_trip_1', 'departure_time')) - as_timestamp(now()) - (wiggle_room_in_minutes * 60))/60 }}`

Template code had a sign the wrong way for anyone else using this.

**Comment by andystewart999** [comment]
OK - I've updated sensor.py to support a new configuration setting ```departure_time_offset``` and to add a new attribute ```offset_departure_time``` which basically is the time the train is expected to leave minus whatever ```departure_time_offset``` is.

```
  - platform: ha_transportnsw
    api_key: <blah>
    origin_id: 207210
    destination_id: 207710
    transport_type: 1 # Only trains
    return_info: medium
    name: "Gordon to Waitara"
    trips_to_create: 1
    trip_wait_time: 2
    departure_time_offset: 11
```

Take the contents of the attached file and paste them over ```homeassistant/custom_components/ha_transportnsw/sensor.py```, then restart Home Assistant.  Each journey will then include the new attribute:

![image](https://github.com/user-attachments/assets/3c321787-47fa-4a7a-8bfa-ddb4bb5bb7e8)

Depending on how you want to actually consume this information though, it does seem to me that just adding your `wiggle room` factor to ```trip_wait_time``` would achieve exactly the same thing, in that it would only show trips that depart, say, 2 + 11 minutes from now?

[sensor.txt](https://github.com/user-attachments/files/17541343/sensor.txt)

**Comment by ktwb** [comment]
Thank you king! Sorry for the delay, I'm still working through this but your suggestions have been very helpful. I'll update this thread once I've finished it all up!

**Comment by andystewart999** [comment]
How did you go?  I'm going to go ahead and close this issue but let me know if you have any other questions.
